sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -39,15 +39,26 @@ case class Project(projectList: Seq[NamedExpression], child: SparkPlan)
     child.asInstanceOf[CodegenSupport].produce(ctx, this)
   }
 
+  override def usedInputs: AttributeSet = {
```

Comment:
Why? I think we need to have stronger semantics so that this is not necessary. I think if each operator just always ensured the referenced attributes were populated at the start of consume, we don't need this.